tools/write_pyi.py

```diff
@@ -99,8 +100,27 @@ def generate_pyi_for_proxy(
                 continue
             meth = getattr(cls, name, None)
             if callable(meth):
+                # check if there are overloads and stub them before the default
+                fn = getattr(cls, name)
+                for ovl in typing.get_overloads(fn):
+                    _generate_stub_for_meth(
+                        ovl,
+                        cls,
+                        name,
+                        printer,
+                        env,
+                        name in context_managers,
+                        True,
```

Comment:
since we start to have a lot of args in `_generate_stub_for_meth`, maybe the last two bool could by passed as kw args?

Reply:
I'm not sure i understand the ask here. 

Are you looking for a signature like:

```
def _generate_stub_for_meth(fn, cls, name, printer, env, **kwargs):
```

or

```
def _generate_stub_for_meth(fn, cls, name, printer, env, is_context_manager = False, is_overload = False):
```

I can probably get rid of the `name` argument and just use `__name__` off of the method.

Reply:
No, I just meant using `is_context_manager=name in context_managers, is_overload=True` when calling the function

not very important either way

Reply:
addressed this in latest push